### PR TITLE
Add a `tail` parameter to pod collector.

### LINKED ISSUE #216: Allow specifying number of lines for `pod` test collector
`kubectl logs` defaults to just 10 trailing lines when a selector is used:
```
      --tail=-1: Lines of recent log file to display. Defaults to -1 with no selector, showing all log lines otherwise
10, if a selector is provided.
```

Please add an optional `tail` field that lets use specify the number of lines (or -1 for all), since 10 is sometimes insufficient.

ref: https://github.com/kudobuilder/kuttl/blob/c5f944beaaa4ecf92c55898fbf35b6b7c634c148/pkg/apis/testharness/v1beta1/collector.go#L115-L116

### PR DESCRIPTION
**Pull request by porridge**
**What this PR does / why we need it**:

Ability to set the number of lines to collect.
Also minor improvements to contributor guide.

Fixes #216

### CODE REVIEW

File: pkg/apis/testharness/v1beta1/collector_test.go [hunk 1]
```diff
@@ -3,6 +3,8 @@ package v1beta1
 import (
 	"strings"
 	"testing"
+
+	"gotest.tools/assert"
```
Comment (kensipe) [inline pkg/apis/testharness/v1beta1/collector_test.go:7]: please switch to testify (which will remove the new dependency as well with a go mod verify)

**Review by kensipe — CHANGES_REQUESTED** [review]
love it!  thanks!

the project has a dependency on testify... could you switch out the newly added dependency.. then we are good!

**Comment by kensipe** [comment]
briefly looked into this... I see you are getting DeepEqual... is there a way to do this with our current tooling?

**Comment by kensipe** [comment]
value of DeepEquals

```
=== RUN   TestPodCommand/pod_name_with_set_tail
    collector_test.go:156: assertion failed: 
        --- cmd.Command
        +++ tt.cmd
          strings.Join({
          	"kubectl logs --prefix foo -n $NAMESPACE --all-containers --tail=",
          	"4",
        - 	"2",
        + 	"3",
          }, "")
        
```

value of testify `.Equal`

```
=== RUN   TestPodCommand/pod_name_with_set_tail
    collector_test.go:157: 
        	Error Trace:	collector_test.go:157
        	Error:      	Not equal: 
        	            	expected: "kubectl logs --prefix foo -n $NAMESPACE --all-containers --tail=42"
        	            	actual  : "kubectl logs --prefix foo -n $NAMESPACE --all-containers --tail=43"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-kubectl logs --prefix foo -n $NAMESPACE --all-containers --tail=42
        	            	+kubectl logs --prefix foo -n $NAMESPACE --all-containers --tail=43
        	Test:       	TestPodCommand/pod_name_with_set_tail
--- FAIL: TestPodCommand (0.00s)

```

File: pkg/apis/testharness/v1beta1/collector_test.go (hunk 1 above)
Comment (porridge) [reply pkg/apis/testharness/v1beta1/collector_test.go]: Awesome, works even better!

**Comment by porridge** [comment]
@kensipe I switched to testify and mentioned it in the contributor guide along with a couple other minor improvements there, PTAL.

**Review by kensipe — APPROVED** [review]
fantastic!  nice work!

**Pull request merged**